CombinatorialGames/Game/IGame.lean

```diff
@@ -1188,128 +1190,165 @@ private def InvTy.val' {x : IGame}
-· igame_wf
+· exact .of_mem_leftMoves ((equivShrink _).symm x).2.1
+· exact .of_mem_rightMoves ((equivShrink _).symm x).2.1
 
 private abbrev InvTy.val (x : IGame) (b : Bool)
-    (i : InvTy (Shrink {y ∈ x.leftMoves | 0 < y}) (Shrink x.rightMoves) b) : IGame :=
+    (i : InvTy (Shrink {y ∈ x.leftMoves | 0 < y}) (Shrink {y ∈ x.rightMoves | 0 < y}) b) : IGame :=
   i.val' (inv' ∘ Subtype.val ∘ (equivShrink _).symm) (inv' ∘ Subtype.val ∘ (equivShrink _).symm) b
 
-/-- The inverse of a game `x = {s | t}ᴵ` is `{s' | t'}ᴵ`, where `s'` and `t'` are the smallest sets
-such that `0 ∈ s'`, and such that `(1 + (z - x) * a) / z, (1 + (y - x) * b) / y ∈ s'` and
-`(1 + (y - x) * a) / y, (1 + (z - x) * b) / z ∈ t'` for `y ∈ s` positive, `z ∈ t`, `a ∈ s'`, and
+/-- The inverse of a positive game `x = {s | t}ᴵ` is `{s' | t'}ᴵ`, where `s'` and `t'` are the
+smallest sets such that `0 ∈ s'`, and such that `(1 + (z - x) * a) / z, (1 + (y - x) * b) / y ∈ s'`
+and `(1 + (y - x) * a) / y, (1 + (z - x) * b) / z ∈ t'` for `y ∈ s` positive, `z ∈ t`, `a ∈ s'`, and
 `b ∈ t'`.
 
-If `x` is a positive numeric game, then `x * x⁻¹ ≈ 1`. The value of this function on any other game
-should be treated as a junk value. -/
+If `x` is negative, we define `x⁻¹ = -(-x)⁻¹`. For any other game, we set `x⁻¹ = 0`.
+
+If `x` is a non-zero numeric game, then `x * x⁻¹ ≈ 1`. The value of this function on any non-numeric
+game should be treated as a junk value. -/
 instance : Inv IGame where
-  inv := inv'
+  inv x := by classical exact if 0 < x then inv' x else if x < 0 then -inv' (-x) else 0
 
 instance : Div IGame where
   div x y := x * y⁻¹
 
+open Classical in
+private theorem inv_eq' {x : IGame} :
+    x⁻¹ = if 0 < x then inv' x else if x < 0 then -inv' (-x) else 0 :=
+  rfl
+
+private theorem inv_eq {x : IGame.{u}} (hx : 0 < x) :
+    x⁻¹ = {.range (InvTy.val x false) | .range (InvTy.val x true)}ᴵ := by
+  rw [inv_eq', if_pos hx, inv']
+  rfl
+
 protected theorem div_eq_mul_inv (x y : IGame) : x / y = x * y⁻¹ := rfl
 
+theorem inv_of_equiv_zero {x : IGame} (h : x ≈ 0) : x⁻¹ = 0 := by simp [inv_eq', h.not_lt, h.not_gt]
+
+@[simp] protected theorem inv_zero : (0 : IGame)⁻¹ = 0 := inv_of_equiv_zero .rfl
 @[simp] protected theorem zero_div (x : IGame) : 0 / x = 0 := zero_mul _
 @[simp] protected theorem neg_div (x y : IGame) : -x / y = -(x / y) := neg_mul ..
 
-private theorem inv_eq (x : IGame.{u}) :
-    x⁻¹ = {.range (InvTy.val x false) | .range (InvTy.val x true)}ᴵ := by
-  change inv' x = _
-  rw [inv']
-  rfl
+@[simp]
+protected theorem inv_neg (x : IGame) : (-x)⁻¹ = -x⁻¹ := by
+  rw [inv_eq', inv_eq']
+  obtain h | h | h | h := lt_or_antisymmRel_or_gt_or_incompRel x 0
+  repeat
+    simp [h, h.asymm]
+    simp [h.not_lt, h.not_gt]
 
 /-- The general option of `x⁻¹` looks like `(1 + (y - x) * a) / y`, for `y` an option of `x`, and
 `a` some other "earlier" option of `x⁻¹`. -/
 @[pp_nodot]
 def invOption (x y a : IGame) : IGame :=
```

Comment:
An annoying consequence of this refactor is that the `InvTy.val` values are no longer def-eq to `invOption`, which is why this section becomes longer.